src/libraries/shims/fakes/Directory.Build.targets

```diff
@@ -0,0 +1,30 @@
+<Project>
+
+  <Import Project="..\..\Directory.Build.targets" />
+
+  <PropertyGroup>
+    <TypeForwardedTypesCompilePath>$(IntermediateOutputPath)$(MSBuildProjectName).TypeForwardedTypes.cs</TypeForwardedTypesCompilePath>
+  </PropertyGroup>
+
+  <Target Name="CreateCompileSourceForTypeForwardedTypes"
+          BeforeTargets="CoreCompile"
+          Inputs="$(MSBuildProjectFullPath)"
+          Outputs="$(TypeForwardedTypesCompilePath)">
+    <Error Text="The 'TypeForwardedType' item must not be empty." Condition="'@(TypeForwardedType)' == ''" />
+
+    <ItemGroup>
+      <TypeForwardedType Namespace="$([System.String]::new('%(Identity)').Substring(0, $([System.String]::new('%(Identity)').LastIndexOf('.'))))" />
+      <TypeForwardedType TypeName="$([System.String]::new('%(Identity)').Substring($([MSBuild]::Add($([System.String]::new('%(Identity)').LastIndexOf('.')), 1))))" />
+      <TypeForwardedTypesLine Include="namespace %(TypeForwardedType.Namespace) { public class %(TypeForwardedType.TypeName) { } }" />
```

Comment:
To avoid accidental consumption you might want to make these `static` or make the constructor private.  That way we can limit the usage to just `typeof` statements.